May use 64+bit CIDs as address validation (#3924)
quicwg/base-drafts: draft-ietf-quic-transport.md

endpoints.  In particular, receipt of a packet protected with Handshake keys
confirms that the client received the Initial packet from the server.  Once the
server has successfully processed a Handshake packet from the client, it can

> consider the client address to have been validated.  Servers MAY treat the

Reviewer: ```suggestion
consider the client address to have been validated.  Servers MAY also treat the
```

> receipt of a packet using a server generated destination connection ID with

Reviewer 2: nit: s/server generated/server-generated/

Author: SGTM, for the future, suggestions are faster for me to commit.

> receipt of a server generated destination connection ID with at least 64 bits

Reviewer 3: ```suggestion
receipt of a packet using a server generated destination connection ID with at least 64 bits
```

> consider the client address to have been validated.  Servers MAY treat the
> receipt of a packet using a server-generated destination connection ID with
> at least 64 bits of entropy as address validation.

Reviewer 4: ```suggestion
consider the client address to have been validated.

Additionally, if a server sends a packet carrying a Source Connection
ID field that has at least 64 bits of entropy to a client address,
and if the server subsequently receives a packet from the client
carrying this server-generated connection ID in its Destination
Connection ID field, the server MAY consider the address to be
validated.
```

Reviewer 5: Longest sentence ever.  Though I agree with having a new paragraph for this separate concept.

> A server MAY consider the client address to be valid if the client uses a connection ID chosen by the server and the connection ID contains at least 64 bits of entropy.

BTW, it is not necessary that the connection ID be provided by the server in Source Connection ID, it's just overwhelmingly likely.  (In an extreme case, involving very specific sorts of loss, this could use a connection ID provided in NEW_CONNECTION_ID; preferred_address is also unlikely but marginally more likely.)

Reviewer 4: Good points, both. I like your sentence better, but I'd propose "Additionally" or "also", as Kazuho proposes.

Author: SG, I tried to update taking into account all your comments/suggestions
